### ISSUE TITLE
Resolve Client-side API issue

### ISSUE THREAD

**Issue by HR** [codexa/firetext#147]
One of the problem that we are encountering a lot is the client-side API integration since we can't don't have access to a client side redirect uri for oauth and non oauth. To unify things I think that we make calls via api.codexa.org (which would be like a private api and a web app to the service) to services such as google drive, skydrive, etc.
This would all be done by our assigned client id. This would make everything more secure and trackable.
So it will be:
Firetext => api.codexa.org => https://accounts.google.com/o/oauth2/auth
Client-side   >   Server-side

**Comment by ferndot** [comment]
Indeed, I am just wondering how we will implement this (no experience with writing server-side apis).

**Comment by HR** [comment]
Maybe Zend Rest Server http://framework.zend.com/manual/1.12/en/zend.rest.server.html

**Comment by HR** [comment]
This server mechanism is going to be private not in public repo because of security risks
